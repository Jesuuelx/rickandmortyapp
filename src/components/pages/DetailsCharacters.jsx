import React, { useEffect } from "react";
import { getOnceCharacter } from "../redux/store/slices/thunks";
import { useParams } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { setItemsWhiteList } from "../redux/store/slices/rickandmorthySlices";

export const DetailsCharacters = () => {
  const { id } = useParams();

  const { character, isLoading, whiteList } = useSelector(
    (state) => state.rickandmorthy
  );

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getOnceCharacter(id));
  }, [id]);

  const handleAdd = () => {
    dispatch(setItemsWhiteList(character));
  };

  return (
    <div>
      {isLoading ? (
        <p className="animate__animated animate__fadeInLeft">Loading...</p>
      ) : (
        <div className="card animate__animated animate__fadeInLeft">
          <p>{character.name}</p>
          <img src={character.image} alt={character.name} />
          <p>GENDER: {character.gender}</p>
          <p>STATUS: {character.status}</p>
          <p>SPECIES: {character.species} </p>
        </div>
      )}

      <button className="outline" onClick={handleAdd}>
        AGREGAR A FAVORITOS
      </button>
    </div>
  );
};
