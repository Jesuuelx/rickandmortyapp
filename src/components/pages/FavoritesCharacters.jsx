import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { remoteWhiteList } from "../redux/store/slices/rickandmorthySlices";

const init = () => {
  return JSON.parse(localStorage.getItem("whiteList")) || [];
};

export const FavoritesCharacters = () => {
  const dispatch = useDispatch();

  const { whiteList } = useSelector((state) => state.rickandmorthy);

  return (
    <>
      <h2> Favorites </h2>
      <div className="card-grid animate__animated animate__fadeInLeft">
        {init().map((item) => (
          <div key={item.name} className="card">
            <div>{item.name}</div>
            <img src={item.image} alt={item.name} />
            <button
              className="outline"
              onClick={() => dispatch(remoteWhiteList(item.name))}
            >
              REMOVE
            </button>
          </div>
        ))}
      </div>
    </>
  );
};
