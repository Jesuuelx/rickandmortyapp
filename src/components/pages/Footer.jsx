import React from "react";
import GithubIcon from "../../assets/svgs/carbon_logo-github.svg";
import Linkedin from "../../assets/svgs/carbon_logo-linkedin.svg";
export const Footer = () => {
  return (
    <footer className="foot-container">
      <div className="foot-logo">NFT</div>
      <div social-container>
        <a href="https://github.com/Jesuuelx">
          <img src={GithubIcon} alt="Github Icon" />
        </a>
        <a href="https://www.linkedin.com/in/jes%C3%BAs-alejandro-salas-teran-78a091220/">
          <img src={Linkedin} alt="Twitter Logo" />
        </a>
      </div>
    </footer>
  );
};
