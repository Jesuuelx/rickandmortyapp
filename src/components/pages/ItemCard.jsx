import React from "react";
import { Link } from "react-router-dom";

export const ItemCard = ({ name, image, id }) => {
  return (
    <div className="card">
      <p>{name}</p>
      <div>
        <img  src={image} alt={image} />
      </div>
  <button className="outline">    <Link to={`/${id}/details`}> Ver Mas...</Link> </button>
    </div>
  );
};
