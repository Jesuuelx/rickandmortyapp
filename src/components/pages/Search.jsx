import React from "react";
import { useForm } from "../../hooks/useForm";
import queryString from "query-string";
import { useNavigate, useLocation } from "react-router-dom";

export const Search = ({ items, setShowItems }) => {
  const navigate = useNavigate();

  const location = useLocation();

  const query = queryString.parse(location.search);

  const { q = "" } = query;

  const [{ seachText }, handleInputChange, reset] = useForm({
    seachText: q,
  });
  const handleSubmit = (e) => {
    e.preventDefault();
    const newItems = items.filter((item) =>
      item.name.toLocaleLowerCase().includes(seachText.toLocaleLowerCase())
    );
    setShowItems([...newItems]);
    navigate(`?q=${seachText}`);
  };

  return (
    <form onSubmit={handleSubmit} className='input-outline' aria-label="form">
      <input
        type="text"
        placeholder="Buscar tu NFT"
        name="seachText"
        autoComplete="off"
        value={seachText}
        onChange={handleInputChange}
      />

      <button type="submit" className="outline">Buscar</button>
    </form>
  );
};
