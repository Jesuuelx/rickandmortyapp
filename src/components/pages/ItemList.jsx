import React, { useEffect, useState } from "react";
import { ItemCard } from "./ItemCard";
import { Search } from "./Search";

export const ItemList = ({ items, isLoading }) => {


  const [showItems, setShowItems] = useState([]);

  useEffect(() => {
    setShowItems(items);
  }, [items]);

  const handleReset = () => {
    setShowItems([...items]);
  };

  return (
    <>
      <div className="banner-cont">
        <div className="banner-logo">
          <h1 className="nft-h1">NFT</h1>
        </div>
        <h2>Rick And Morty Coleccion NFT</h2>
        <p>Unica Coleccion NFT de Rick And Morty</p>
        <div className="rect-group">
          <div className="rect">------</div>
        </div>
      </div>

      <Search items={items} setShowItems={setShowItems} />
      <hr />

      {showItems.length === 0 && <p>Busca Otro Personaje</p>}
      <button className="outline" onClick={handleReset}>
        {" "}
        RESET{" "}
      </button>
      <hr />
      {isLoading && <p className="animate__animated animate__fadeInLeft">Loading</p>}

      <div className="card-grid animate__animated animate__fadeInLeft">
        {showItems.map((item) => (
          <ItemCard key={item.id} {...item} />
        ))}
      </div>
    </>
  );
};
