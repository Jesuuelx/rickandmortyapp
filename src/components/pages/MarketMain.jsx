import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { getCharacters } from "../redux/store/slices/thunks";
import { ItemList } from "./ItemList";

export const MarketMain = () => {
  const dispatch = useDispatch();

  const { items, isLoading } = useSelector((state) => state.rickandmorthy);

  useEffect(() => {
    dispatch(getCharacters());
  }, []);

  return <ItemList items={items} isLoading={isLoading} />;
};
