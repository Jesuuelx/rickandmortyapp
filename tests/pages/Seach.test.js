import {  render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { Search } from '../../src/components/pages/Search';

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUseNavigate,
}));



describe('Pruebas en <Search />', () => {


    beforeEach(() => jest.clearAllMocks() );

    
    test('debe de mostrarse correactamente con valores por defecto', () => {
        
        const { container } =render(
            <MemoryRouter>
                <Search />
            </MemoryRouter>
        );
        expect( container ).toMatchSnapshot();
        
    });


    test('debe de mostrar a Batman y el input con el valor del queryString', () => {
        
        render(
            <MemoryRouter initialEntries={['/search?q=rick']}>
                <Search />
            </MemoryRouter>
        );
        
        const input = screen.getByRole('textbox');
        expect( input.value ).toBe('rick');
        
       

      
        
    });

  


})