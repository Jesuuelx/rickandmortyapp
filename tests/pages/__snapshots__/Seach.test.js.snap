// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage /> debe de mostrarse correactamente con valores por defecto 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      autocomplete="off"
      name="seachText"
      placeholder="Buscar tu NFT"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Buscar
    </button>
  </form>
</div>
`;
